### PR TITLE
Build/vite modes

### PR DESCRIPTION
**Pull request by lukeed**
* adjust `exports` map for named entries
* replace `deep/path/to/file.js` with named entries
* update `vite` (hence large lock file diff)

---

// cc @JoviDeCroock

### CODE REVIEW

File: packages/vite/src/utils.js [hunk 1]
```diff
@@ -1 +1,2 @@
-export { compareSignatures } from '@prefresh/utils';
+import { compareSignatures as compare } from '@prefresh/utils';
```
Comment (lukeed) [inline packages/vite/src/utils.js:1]: This is quick workaround for bundt – didn't realize I was missing `export { ... } from ...` support.

It's effectively the same thing anyway, for the time being.

File: packages/vite/package.json [hunk 2]
```diff
@@ -1,18 +1,36 @@
...
+			"require": "./runtime/index.js"
+		},
+		"./utils": {
+			"import": "./utils/index.mjs",
+			"require": "./utils/index.js"
 		},
 		"./package.json": "./package.json",
 		"./": "./"
 	},
+	"modes": {
```
Comment (lukeed) [inline packages/vite/package.json:22]: This is for `bundt` – without it it'd default to `src/index.js` only, but _with it_ it produces multiple targets:

```
modes: {
  "<target directory>": "<entry file>"
}
```

> Note: `default` is special

For this package setup, having a `main` and `module` key means that each "mode" produces its own `index.js` and `index.mjs` file.

**Review by JoviDeCroock — APPROVED** [review]
Awsesome, this looks good to me! Quite sad that this has to happen to support yarn 2 and pnpm. I'll do the other integrations then as well 😅

**Comment by lukeed** [comment]
Unfortunately it has to be done for Node.js in general now. The whole reason bundt has this multi-mode support is because having single-file modules all the time isn't realistic, and given `exports` map's existence, we're going to have to do this weird, multi-format config-shuffle for quite some time.

**Pull request merged**